### PR TITLE
TS fixed their decorators-renaming bug in 3.8-beta, so upgraded all p…

### PR DESCRIPTION
**Pull request by deregtd**
…ackages to latest versions.  Addressed a bunch of lint that appeared with the update.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,40 +22,41 @@
     "tslib": "^1.10.0"
   },
   "peerDependencies": {
-    "react": "^16.11.0"
```
Comment (berickson1) [inline package.json]: I don't think this should be updated - resub works just fine with react 16.11, there's no real dep on 16.12 & peerDeps are managed by the consumer

File: package.json [hunk 2]
```diff
@@ -22,40 +22,41 @@
...
-    "rimraf": "3.0.0",
+    "react": "^16.12.0",
+    "react-dom": "^16.12.0",
+    "rimraf": "3.0.1",
     "ts-loader": "6.2.1",
-    "tslint": "5.20.0",
-    "typescript": "3.5.3",
-    "webpack": "4.41.2"
+    "tslint": "6.0.0",
+    "typescript": "3.8.0-dev.20200201",
```
Comment (berickson1) [inline package.json]: why dev build of 3.8?
Use 3.7.5 or 3.8.0-beta

File: test/AutoSubscribe.spec.tsx [hunk 3]
```diff
@@ -7,6 +7,7 @@
  */
 
 import * as React from 'react';
+// eslint-disable-next-line
```
Comment (berickson1) [inline test/AutoSubscribe.spec.tsx]: Why is this needed?

Comment (deregtd) [reply test/AutoSubscribe.spec.tsx]: awkward previous code :(

File: package.json (hunk 2 above)
Comment (deregtd) [reply package.json]: https://github.com/microsoft/TypeScript/issues/33269 only landed a few days ago so it's not in 3.8.0-beta
